launcher/common/launcher.py

```diff
@@ -0,0 +1,142 @@
+            return common.enums.LauncherReturnCodes.PARAMETER_ERROR
+
+        ########
+        # STEP 2 - Setting Up the Configuration Manager
+        ########
+        # TO BE DONE: __logs_root_dir should be set based on environment variable or by using another mechanism
+        # e.g. self.__logs_root_dir = os.environ['HOME'] + '/co_sim/logs'
+        self.__configuration_manager = configurations_manager.ConfigurationsManager()
+        self.__logger = self.__configuration_manager.load_log_configurations(name='launcher')
+        self.__logger.info('START: Co-Simulation Launcher')
+
+        ########
+        # STEP 3 - Co-Simulation Plan
+        ########
+        self.__co_sim_plan_validator = \
+            common.xml_managers.CoSimulationPlanXmlManager(configuration_manager=self.__configuration_manager,
+                                                           logger=self.__logger,
+                                                           xml_filename=self.__args.action_plan)
+
+        # STEP 3.1 - Validating Co-Simulation Plan XML file
+        if not self.__co_sim_plan_validator.dissect() == common.enums.XmlManagerReturnCodes.XML_OK:
+            return common.enums.LauncherReturnCodes.XML_ERROR
+
+        # STEP 3.2 - Getting the parameters found on the Co-Simulation Plan XML file
+        self.__action_plan_parameters_dict = self.__co_sim_plan_validator.get_parameters_dict()
+
+        # STEP 3.3 - Getting the variables found on the Co-Simulation Plan XML file
+        self.__action_plan_variables_dict = self.__co_sim_plan_validator.get_variables_dict()
+
+        ###################################
+        # STEP 4 Co-Simulation Parameters #
+        ###################################
+        self.__co_sim_parameters_validator = \
+            common.xml_managers.CoSimulationParametersXmlManager(configuration_manager=self.__configuration_manager,
+                                                                 logger=self.__logger,
+                                                                 xml_filename=self.__args.parameters)
+
+        # STEP 4.1 - Validating Co-Simulation Parameters XML file
+        if not self.__co_sim_parameters_validator.dissect() == common.enums.XmlManagerReturnCodes.XML_OK:
+            return common.enums.LauncherReturnCodes.XML_ERROR
+
+        # STEP 4.2 - Getting the parameters found on the Co-Simulation Plan XML file
+        self.__parameters_parameters_dict = self.__co_sim_parameters_validator.get_parameters_dict()
+
+        # STEP 4.3 - Getting the variables found on the Co-Simulation Plan XML file
+        self.__parameters_variables_dict = self.__co_sim_parameters_validator.get_variables_dict()
+
+        # STEP 5 - Converting Co-Simulation XML parameters into JSON
+        if not self.generate_parameters_json_file() == common.enums.LauncherReturnCodes.OK:
+            return common.enums.LauncherReturnCodes.JSON_FILE_ERROR
+
+        # STEP 6 - Launching action plan
```

Comment:
Nicely commented. So here would be for example the execution of Lionels use case? We could discuss then the connection to my refactor work and what I can prepare and provide to fill this TODO !

Reply:
As has been mentioned,  **TVB-NEST/tests/test_translator_nest_to_tvb.sh** is the selected test case. From the perspective of this launcher, we must just add the reference to the script or scripts where your improvement have been placed, and making sure that they will be taken into account by the action_plan.xml or the action.xml it self.

The idea is what the launcher should "launch", it must be based on the .XML files passed as command-line parameters.